### PR TITLE
Added contribution guidelines

### LINKED ISSUE #47: Make contribution guidelines more detailed

### PR DESCRIPTION
**Pull request by isaack-njama**
fixes #47

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+origin  git@github.com:[your_github_handle]/mintgate.git (fetch)
+origin  git@github.com:[your_github_handle]/mintgate.git (push)
+upstream        git@github.com:GETLN/mintgate.git (fetch)
+upstream        git@github.com:GETLN/mintgate.git (push)
+```
+7. To sync your fork with the latest changes on the `upstream/main` branch:
+```bash
+$ git checkout main
+$ git fetch upstream
+$ git merge upstream/main
```
Comment (okjodom) [inline CONTRIBUTING.md:38]: `git rebase upstream/main` - we prefer linear history!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+10. Create a feature branch for your changes:
+```bash
+git checkout -b feature_name
+```
+- It is good practice to always use a feature branch. Working directly from the **main** branch is risky business!
+11. When done editing, add and commit to record your changes in Git:
+```bash
+$ git add modified_files
+$ git commit -m "brief description of changes"
```
Comment (okjodom) [inline CONTRIBUTING.md:71]: We should highly encourage [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)

**Review by okjodom — APPROVED** [review]

**Comment by okjodom** [comment]
@isaack-njama this is a fantastic contribution. I approved with suggestions, but waiting on your decision whether you want to push an update or make a quick follow up PR

File: CONTRIBUTING.md (hunk 1 above)
Comment (isaack-njama) [reply CONTRIBUTING.md:38]: Noted, i'll go ahead and make that change.

**Comment by isaack-njama** [comment]
> @isaack-njama this is a fantastic contribution. I approved with suggestions, but waiting on your decision whether you want to push an update or make a quick follow up PR

Currently working on making an update to have contributors make [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).

File: CONTRIBUTING.md (hunk 2 above)
Comment (isaack-njama) [reply CONTRIBUTING.md:71]: Learnt a few things myself. Hope the examples are intuitive.

Comment (okjodom) [reply CONTRIBUTING.md:71]: Ack. Looks good

**Pull request merged**